### DISCUSSION TITLE
Reading query from project instead of via Admin UI

### DISCUSSION THREAD

**Discussion by ns8482e** [OrchardCMS/OrchardCore#10563] [Q&A] [answered]
Is it possible to read query from file system(embedded in module) instead of Document manager?

**Comment by deanmarcussen** [comment, marked as answer]
I actually did this recently for some graphql queries I preferred to embed.

Implemented it like recipe harvestors, but harvested `.graphql` instead.

I chose not the wrap the query manager, but implemented another one, as they were not for front end queries.

But you could wrap the query manager and either divert to the document based or embedded ones.

It was quite effective, as it was easy to cut and paste queries out of graphql explorers on the site we were connecting to.

I think I parsed the query name out of the graphql file.
